### PR TITLE
Tests and optimizes: rocket_add_url_protocol()

### PR DESCRIPTION
**Pull request by hellofromtonya**
`rocket_add_url_protocol()`:

- Adds unit tests
- Adds integration tests
- Optimizes by:
    - flipping http/https checks into guard clause
    - returning URL from `set_url_scheme()` instead of assigning to variable and then returning

Tests were done _before_ the code was optimized to ensure the changes did not change the behavior.

### CODE REVIEW

File: inc/functions/formatting.php [hunk 1]
```diff
@@ -259,22 +259,29 @@ function rocket_remove_url_protocol( $url, $no_dots = false ) {
 }
 
 /**
- * Add HTTP protocol to an url that does not have
+ * Add HTTP protocol to an url that does not have.
```
Comment (Tabrisrp) [inline inc/functions/formatting.php:262]:
```suggestion
 * Add HTTP protocol to an url that does not have it.
```

File: inc/functions/formatting.php [hunk 2]
```diff
@@ -259,22 +259,29 @@ function rocket_remove_url_protocol( $url, $no_dots = false ) {
...
+ * @return string $url The URL with protocol.
  */
 function rocket_add_url_protocol( $url ) {
+	// Bail out if the URL starts with http:// or https://.
+	if (
+		strpos( $url, 'http://' ) !== false
+		||
+		strpos( $url, 'https://' ) !== false
+	) {
+		return $url;
```
Comment (Tabrisrp) [inline inc/functions/formatting.php:277]: The URL should still go through `set_url_scheme()` to make sure it uses the correct scheme depending on the value of `is_ssl()`

Comment (hellofromtonya) [reply inc/functions/formatting.php:277]: It doesn't currently. Currently, it only goes through `set_url_scheme()` to get the `http:` or `http:` protocol if the given URL doesn't have either.

Comment (hellofromtonya) [reply inc/functions/formatting.php:277]: @Tabrisrp Are you thinking this is a bug?

Comment (Tabrisrp) [reply inc/functions/formatting.php:277]: oh yes you're right, I read the original wrong, sorry about that!

I think it's ok that way in fact, looking at where it's used in our codebase, there is no issue with it.

**Pull request merged**